### PR TITLE
Extra Notes Added in README.md

### PR DESCRIPTION
**Pull request by Sagar73594**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,4 +54,9 @@ Finally, if you are sure no one else had the issue, follow the [ISSUE_TEMPLATE](
 
 ### Tip
 
-Sometimes to program ESP32 via serial you must keep GPIO0 LOW during the programming process
+1 > Sometimes to program ESP32 via serial you must keep GPIO0 LOW during the programming process
+
+2 > On some ESP32 development boards, When you trying to upload a new sketch, the Arduino IDE gives you a fatal error. ( like A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header) This means your ESP32 board does not goes to automatically into flashing/uploading mode. 
+
+Solution:  To make flashing automatically, You have to connect a 10 uF electrolyte capacitor between the EN pin and GND pin.
```
Comment (pedrominatel) [inline README.md:61]: Solution: To make flashing automatically, you have to connect a 10 uF capacitor between the EN and GND pins. This capacitor value could be lower (> 2.2 uF) if you don't have this part available.

EN <----||---- GND

The capacitor can be electrolytic, ceramic or tantalum. Using lower values like 2.2uF and 4.7 uF is also possible in most of the cases.

File: README.md [hunk 2]
```diff
@@ -54,4 +54,9 @@ Finally, if you are sure no one else had the issue, follow the [ISSUE_TEMPLATE](
 
 ### Tip
 
-Sometimes to program ESP32 via serial you must keep GPIO0 LOW during the programming process
+1 > Sometimes to program ESP32 via serial you must keep GPIO0 LOW during the programming process
+
+2 > On some ESP32 development boards, When you trying to upload a new sketch, the Arduino IDE gives you a fatal error. ( like A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header) This means your ESP32 board does not goes to automatically into flashing/uploading mode.
```
Comment (pedrominatel) [inline README.md:59]: 2 > In some ESP32 development boards, when trying to upload a new sketch, the Arduino IDE gives you a fatal error (i.e *A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header*).
This means your ESP32 board is not automatically in the flashing/uploading mode.

**Review by pedrominatel — COMMENTED** [review]
Here are my comments.

**Comment by pedrominatel** [comment]
Hi @Sagar73594, can you please validate my comments? If they are relevant, please modify the file and squash the commits.

Thanks!

File: README.md (hunk 2 above)
Comment (Sagar73594) [reply README.md]: Good to know, Some brief comments. All changes are added in README.md.
Thank You.

**Comment by Sagar73594** [comment]
> Hi @Sagar73594,
> 
> I would like to suggest a different approach about this change.
> 
> First, we can add a reference for the [esptool troubleshooting](https://github.com/espressif/esptool#troubleshooting).
> This troubleshooting covers different issues you might have during the programming steps.
> 
> Second, we can add the [esp32_hardware_design_guidelines_en.pdf](https://www.espressif.com/sites/default/files/documentation/esp32_hardware_design_guidelines_en.pdf) about the capacitor on the EN pin inside the esptool troubleshooting.
> 
> See chapter **_2.2 - Power-on Sequence and System Reset_**
> image
> 
> This information is relevant not only for Arduino, but also for any ESP32 project (ESP-IDF, Zephyr, NuttX, Micropython...).

Okay, That also great.

**Comment by pedrominatel** [comment]
Hello @Sagar73594 

We are working on the documentation and this information is relevant depending on the development board.
The suggestion is to add it in the "Boards" section in the documentation according to the development that requires this solution.

I'll close this PR and you can open a new one in docs, as soon as the structure is done.

**Comment by Sagar73594** [comment]
Okay. That's great. I'm planning to add more solutions/troubleshooting for different types of boards. I will see the documentation when it's completed.

**Pull request closed without merging**